rococo/repositories/mongodb/mongodb_repository.py

```diff
@@ -0,0 +1,151 @@
+        """Method to convert VersionedModel instance to a data dictionary that can be inserted in MongoDB"""
+        super()._process_data_before_save(instance)
+        data = instance.as_dict(convert_datetime_to_iso_string=False, convert_uuids=True)
+        for field in fields(instance):
+            if data.get(field.name) is None:
+                continue
+
+            field_value = data[field.name]
+
+            if field.metadata.get('field_type') in ['entity_id', 'uuid']:
+                if isinstance(field_value, VersionedModel):
+                    field_value = str(field_value.entity_id).replace('-', '')
+                elif isinstance(field_value, dict):
+                    field_value = str(field_value.get('entity_id')).replace('-', '')
+                elif isinstance(field_value, str):
+                    field_value = field_value.replace('-', '')
+            if isinstance(field_value, UUID):
+                field_value = str(field_value).replace('-', '')
+            if isinstance(field_value, datetime):
+                field_value = field_value.strftime('%Y-%m-%d %H:%M:%S')
+
+            data[field.name] = field_value
+        return data
+
+    def get_move_entity_to_audit_table_query(self, data):
+        instance = self.get_one(self.table_name, "entity_id", {'entity_id': data.get('entity_id')})
+        self.db[f"{self.table_name}_audit"].insert_one(instance)
+    
+    def get_save_query(self, data):
+        self.db[f"{self.table_name}"].get_one_and_update(
+            {'entity_id': data.get("entity_id")},
+            {'$set': data},
+        )
+    
+    def save(self, instance: VersionedModel, send_message: bool = False):
+        """Save func"""
+        data = self._process_data_before_save(instance)
+        with self.adapter:
+            self.adapter.run_transaction([lambda: self.get_move_entity_to_audit_table_query(data), lambda: self.get_save_query(data)])
+            if send_message:
+                # This assumes that the instance is now in post-saved state with all the new DB updates
+                message = json.dumps(instance.as_dict(convert_datetime_to_iso_string=True))
+                self.message_adapter.send_message(self.queue_name, message)
+        return instance
+
+    def _insert(self, data: Dict, collection_name: str):
+        self.db[collection_name].insert_one(data)
+
+    def _outdate(self, data: Dict, collection_name: str):
+        collection = self.db[collection_name]
+        return collection.find_one_and_update(
+            {'entity_id': data.get("entity_id"), 'latest': True, 'active': True},
+            {'$set': {'latest': False}},
+            upsert=False,
+            return_document=ReturnDocument.AFTER
+        )
+
+    def create(self, data: Dict, collection_name: str):
+        return self.update(data, collection_name)
+
+    def create_many(self, data: List[Dict], collection_name: str):
+        self.db[collection_name].insert_many(documents=data)
```

Comment:
Is `create_many` going to be used temporarily, to save not yet migrated to VersionModel models?
Otherwise it has to take care of the version and other fields, similar to save(..)